orientjs/orientjs.d.ts

```diff
@@ -0,0 +1,500 @@
+            list(): Promise<Property[]> & Promise<any[]>;
+            create(config: string, reload?: boolean): Promise<Property> & Promise<any>;
+            create(config?: any, reload?: boolean): Promise<Property> & Promise<any>;
+            create(...config: any[]): Promise<Property[]> & Promise<any>;
+            get(name: string): Promise<Property> & Promise<any>;
+            update(config: any, reload?: boolean): Promise<Property> & Promise<any>;
+            drop(name: string): Promise<Class>;
+            alter(name: string, setting?: any): Promise<Class> & Promise<any>;
+            rename(oldName: string, newName: string): Promise<any>;
+        }
+
+        interface Class {
+
+            db: Db;
+            name: string;
+            shortName: string;
+            defaultClusterId: any;
+            superClass: string;
+            originalName: string;
+            clusterIds: number[];
+
+            configure(config?: any): void;
+            list(limit: number, offset: number): Promise<Class[]> & Promise<any> & Promise<any[]>;
+            list(config: any): Promise<Class[]> & Promise<any> & Promise<any[]>;
+            list(refresh: boolean): Promise<Class[]> & Promise<any> & Promise<any[]>;
+            find(attributes: any, limit: number, offset: number): Promise<any[]>;
+            create(record: Record): Promise<Record> & Promise<any>;
+            create(name: string, parentName?: string, cluster?: string, isAbstract?: boolean): Promise<Class> & Promise<any>;
+            update(cls: any, reload: boolean): Promise<Class> & Promise<any>;
+            reload(): Promise<Class[]>;
+            drop(name: string): Promise<Db>;
+            get(name: string, refresh?: boolean): Promise<Class> & Promise<any>;
+            cacheData(classes: Class[] & any[]): Db;
+            property: Property;
+
+        }
+
+        interface Cluster {
+            name: string;
+            location: string;
+            list(refresh?: boolean): Promise<any[]>;
+            create(name: string, location?: string): Promise<Cluster> & Promise<any>;
+            get(nameOrId: string, refresh?: boolean): Promise<Cluster> & Promise<any>;
+            get(nameOrId: number, refresh?: boolean): Promise<Cluster> & Promise<any>;
+            getByName(name: string, refresh?: boolean): Promise<Cluster> & Promise<any>;
+            getById(id: string, refresh?: boolean): Promise<Cluster> & Promise<any>;
+            drop(name: string): Promise<Db>;
+            count(name: string): Promise<Number>;
+            range(name: string): Promise<any>;
+            cacheData(clusters: Cluster[] & any[]): Db;
+
+        }
+
+        interface RecordMeta {
+            "@rid": string|RID;
+            "@version": string|number;
+           
+        }
+
+        interface Record {
+            rid: RID;
+            create(record: any, options?: any): Promise<Record> ;
+            get(record: Record & any, options?: any): Promise<Record | Buffer>;
```

Comment:
`any` subsumes any other type, so `Record & any` is just `any`, and so is `Record | any`. if `any` is allowed then just use `any` every where.